third_party/move/move-vm/runtime/src/runtime_ref_checks.rs

```diff
@@ -1249,6 +1279,7 @@ impl RefCheckState {
         Self {
             shadow_stack: Vec::new(),
             frame_stack: Vec::new(),
+            in_borrow_field_mut_api: None,
```

Comment:
Leaving a general comment here, so it is easier to have a thread.

Do we have `.masm` support for attributes so that we can test things here? Can you also add a test?

Reply:
Initially I was thinking of reviewing step 3 and 4 separately and then merge 4 back to 3 once both of them are approved.